mmv1/third_party/terraform/resources/resource_google_project.go

```diff
@@ -605,25 +618,32 @@ func enableServiceUsageProjectServices(services []string, project, userAgent str
 	return waitForServiceUsageEnabledServices(services, project, userAgent, config, timeout)
 }
 
-func doEnableServicesRequest(services []string, project, userAgent string, config *Config, timeout time.Duration) error {
+func doEnableServicesRequest(services []string, project, billingProject, userAgent string, config *Config, timeout time.Duration) error {
 	var op *serviceusage.Operation
-
+	var call ServicesCall
 	err := retryTimeDuration(func() error {
 		var rerr error
 		if len(services) == 1 {
 			// BatchEnable returns an error for a single item, so just enable
 			// using service endpoint.
 			name := fmt.Sprintf("projects/%s/services/%s", project, services[0])
 			req := &serviceusage.EnableServiceRequest{}
-			op, rerr = config.NewServiceUsageClient(userAgent).Services.Enable(name, req).Do()
+			call = config.NewServiceUsageClient(userAgent).Services.Enable(name, req)
 		} else {
 			// Batch enable for multiple services.
 			name := fmt.Sprintf("projects/%s", project)
 			req := &serviceusage.BatchEnableServicesRequest{ServiceIds: services}
-			op, rerr = config.NewServiceUsageClient(userAgent).Services.BatchEnable(name, req).Do()
+			call = config.NewServiceUsageClient(userAgent).Services.BatchEnable(name, req)
+		}
+		if config.UserProjectOverride && billingProject != "" {
+			call.Header().Add("X-Goog-User-Project", billingProject)
 		}
+		op, rerr = call.Do()
 		return handleServiceUsageRetryableError(rerr)
-	}, timeout, serviceUsageServiceBeingActivated)
+	},
+		timeout,
+		serviceUsageServiceBeingActivated
+	)
 	if err != nil {
 		return errwrap.Wrapf("failed to send enable services request: {{err}}", err)
 	}
```

Comment:
I think we need to attach the `billing_project` to operation GET requests right below, and `listCurrentlyEnabledServices` list GET requests as well, right?

Reply:
Riley, you are totally right here.

In this iteration I worked from the bottom up instead of top down to ensure every instance of `NewServiceUsageClient` is passed the correct headers. Then chained up resolve the callers from there. Apologies.

Reply:
I believe the `project` value passed in on L657 (right side) is a billing project, so that should actually get set to `billingProject` based on the same incantation's logic in my other comment.